state/machine.go

```diff
@@ -2181,6 +2172,16 @@ func (m *Machine) SetUpgradeCharmProfile(appName, chURL string) error {
 	return nil
 }
 
+func (m *Machine) SetUpgradeCharmProfileOp(appName, chURL string) txn.Op {
+	return txn.Op{
+		C:      machinesC,
+		Id:     m.doc.DocID,
+		Assert: bson.D{{"life", Alive}},
+		Update: bson.D{{"$set", bson.D{{"upgradecharmprofilecharmurl", chURL},
+			{"upgradecharmprofileapplication", appName}}}},
```

Comment:
I wonder if we should be also setting `upgradecharmprofilecomplete` to empty?

Reply:
good idea, added.